### DISCUSSION TITLE
Subdomain for development/staging issue

### DISCUSSION THREAD

**Discussion by AaronMcCloskey** [vercel/vercel#7952] [Help]
I need a way to manually deploy any changes from my `development` branch to `dev.<website>.com` and I'm having some issues.

For what it's worth, I'm using CI/CD via Github Actions (I know Vercel allows for deployment on branch changes, but using Github actions is required for this project), which is why I need to know how to manually deploy. Like `vercel --prod` deploys the production environment, I just have no idea how to target a specific `Preview` environment, or if that's even how I go about what I am trying to achieve?

Here's what I have done so far;

I've added the subdomain to vercel via `Settings > Domain` and connected to a branch

[![enter image description here][1]][1]

[![enter image description here][2]][2]

I've then pointed the CNAME to my Cloudflare domain

[![enter image description here][3]][3]

Then in `Settings > Environment Variables`

I've added environment variables specifically for the `Preview` Environment and selected a custom branch called `development` (same as the `Git Branch` for the sub domain).

[![enter image description here][5]][5]

When I run `vercel` on my `development` branch it attempts to deploy to `Preview` & not `Preview (development)` so it doesn't seem to pick up my environment variables

[![enter image description here][4]][4]

<img width="403" alt="Screenshot 2022-06-12 at 00 32 03" src="https://user-images.githubusercontent.com/20436343/173208233-7847a34d-eff2-4694-b07f-cc4d2492788c.png">

And `dev.<website>.com` seems to be pulling code from my `main` branch and not `development`, as in it's what's on production and not development.

Where have I gone wrong here?

  [1]: https://i.stack.imgur.com/2w6Th.png
  [2]: https://i.stack.imgur.com/eRiz2.png
  [3]: https://i.stack.imgur.com/CLlsc.png
  [4]: https://i.stack.imgur.com/pq9tA.png
  [5]: https://i.stack.imgur.com/VQxd6.png

**Comment by stub4ever** [comment]
I am facing the same problem. Does anyone have an idea?

**Comment by amyegan** [comment]
I'm trying to reproduce the problem, but I have not been able to. I set up a `test` branch with a subdomain, like test.example.com, and an environment variable specific to that branch. I tried deploying the same code twice. Once from GitHub with the automated [Vercel for GitHub](https://vercel.com/docs/concepts/git/vercel-for-github) integration, and a second time from local with `vercel` CLI. Both succeeded and displayed the expected env var value.

<img width="643" alt="image" src="https://user-images.githubusercontent.com/8247105/190489077-a25728ad-2275-48f6-a17d-8291d912db6d.png">

Are you able to share a Deployment ID, [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), or any additional information that might help me replicate the problem?
